mailpoet/lib/Services/AuthorizedSenderDomainController.php

```diff
@@ -23,37 +29,101 @@ public function __construct(
     $this->dmarcPolicyChecker = $dmarcPolicyChecker;
   }
 
+  /**
+   * Get the most recent cached record of Bridge::getAuthorizedSenderDomains
+   */
+  public function getDomainRecords($domain = ''): array {
+    if ($domain) {
+      return $this->currentRecords[$domain] ?? [];
+    }
+    return $this->currentRecords;
+  }
+
   /**
    * Get all Authorized Sender Domains
    *
    * Note: This includes both verified and unverified domains
    */
   public function getAllSenderDomains(): array {
-    $records = $this->bridge->getAuthorizedSenderDomains();
-    $domains = array_keys($records);
-    return $domains;
+    $records = $this->currentRecords = $this->bridge->getAuthorizedSenderDomains();
```

Comment:
Yep. This is done because of caching. `$this->currentRecords` is also used in `AuthorizedSenderDomainController::getDomainRecords`. But with this implementation, the method `getDomainRecords` will not work properly if the cache is not prepopulated. I'll refactor it a bit.